src/djcytoscape/signals.py

```diff
@@ -0,0 +1,63 @@
+from django.dispatch import receiver
+from django.db.models.signals import post_save, post_delete
+
+from djcytoscape.models import CytoScape
+from siteconfig.models import SiteConfig
+from badges.models import Badge
+from quest_manager.models import Quest
+from courses.models import Rank
+from prerequisites.models import Prereq
+
+from djcytoscape.tasks import regenerate_map
+
+
+def regenerate_related_maps(instance):
+    """ Helper function for models that can exist as CytoElements.
+    Attach to a models [post_save, pre_delete] signal to regenerate all related maps when
+    said model is updated or deleted.
+    """
+    if not SiteConfig.get().map_auto_update:
+        return
+
+    # get related maps
+    related_map_ids = list(CytoScape.objects.get_related_maps(instance).values_list('id', flat=True))
+    if not related_map_ids:
+        return
+
+    # run task in background
+    regenerate_map.apply_async(args=[related_map_ids], queue='default')
+
+
+@receiver([post_save, post_delete], sender=Badge)
```

Comment:
This is a way better implementation.  I think we can simplify it even more by adding mutiple senders to a single method like this:

```
@receiver([post_save, post_delete], sender=Quest)
@receiver([post_save, post_delete], sender=Badge)
@receiver([post_save, post_delete], sender=Rank)
def regenerate_related_maps(instance):
    ...
```

You can probably leave the Prereq one as is since it's gotta do some extra stuff.

Reply:
These can go directly on the helper function, no need for an intermediary.

Reply:
done